mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+        and .xtc files for trajectory. It will default to using the tpr file
+        available.
+
+    *solvents*
+        Solvents from directory given to the new instance. Default
+        :code:`solvents=('water', 'octanol')`
+
+    *topology_paths*
+        Specifies topologies used in loading simulated systems. Given
+        with a dictionary with keys-value pair for each solvent and
+        its respective topology path.
+
+    *interactions*
+        Interactions from directory given to the instance. Default
+        :code:`interactions=('Coulomb', 'VDW')`
+
+    .. rubric:: Examples
+
+
+    Typical workflow for MDPOW directory::
+
+        ens = Ensemble(dirname='molecule')
+
+    Typical workflow for adding universes individually::
+
+        ens = Ensemble()
+        ens.add_system(topology='md.gro', trajectory='md.xtc')
+
+    Topology paths can be specified when defining the _ensemble
+    by giving the paths to each solvent topology in a dictionary
+    with the topology_paths argument::
+
+        ens = Ensemble(dirname='molecule', topology_paths={'water': water_path,
+                                                           'octanol': octanol_path}
+
+    Interactions can also be specified when initializing the with
+    a list using the interactions argument::
+
+        ens = Ensemble(dirname='molecule', interactions=['Coulomb']
+
+    .. versionadded:: 0.8.0
+    """
+
+    def __init__(self, dirname=None, solvents=('octanol', 'water'),
+                 topology_paths=None, interactions=('Coulomb', 'VDW')):
+        self.top_dict = topology_paths
+        self._num_systems = 0
+        self._ensemble = {}
+        self._keys = []
+        self._solvents = solvents
+        self._interactions = interactions
+
+        if dirname is None:
+            return
+
+        if not os.path.exists(dirname):
+            logger.error(f"Directory {dirname} does not exist")
+            raise FileNotFoundError
+
+        self._ensemble_dir = dirname
+        self._build_ensemble()
+
+    def __repr__(self):
+        return f"<Ensemble Containing {self._num_systems} System>"
+
+    def __len__(self):
+        return self._num_systems
+
+    def __getitem__(self, index):
+        """Allows dictionary like indexing"""
+        return self._ensemble[index]
+
+    @staticmethod
+    def _sort_trajectories(trajectories: list):
+        """sort list of trajectory files alphabetically"""
+        return sorted(trajectories)
+
+    @staticmethod
+    def _sort_topologies(topologies: list):
+        """sorts list of trajectory files with .tpr first"""
+        top = []
+        logger.info('If more than one topology is present the tpr will be the one used')
+        for i in range(len(topologies)):
+            file = topologies[i]
+            if file.endswith('.tpr'):
+                topologies.pop(i)
+                top = [file] + topologies
+                break
+        return top
+
+    def _load_universe_from_dir(self, solvent):
```

Comment:
Something like
```python
def _load_universe_from_dir(self, solvent=None, **universe_kwargs):
```
where `universe_kwargs` are any kwargs for MDAnalysis.Universe.

Reply:
done